apps/dsb-client-gateway-api/src/app/modules/topic/service/topic.service.ts

```diff
@@ -149,4 +150,34 @@ export class TopicService {
       );
     }
   }
+
+  public async getTopicHistoryByIdAndVersion(id: string, versionNumber: string): Promise<PostTopicDto> {
+
+    const topic: TopicEntity = await this.wrapper.topicRepository.findOne({
+      where: { id, version: versionNumber }
+    });
+
+    return {
+      id: topic.id,
+      name: topic.name,
+      schemaType: topic.schemaType as SchemaType,
+      schema: JSON.stringify(topic.schema),
+      version: topic.version,
+      owner: topic.owner,
+      tags: topic.tags
+    };
+  }
+
+  public async getTopicHistoryById(id: string): Promise<PaginatedTopicResponse> {
+    const [topics, allCount] = await this.wrapper.topicRepository.findAndCount({
+      where: { id }
+    });
+
+    return {
+      count: allCount,
+      limit: 0,
```

Comment:
why limit and page hardcoded?

Reply:
that for return all

Reply:
so do we need to return paginated response then @kim-energyweb ?

Reply:
Yes

Reply:
However, why this part is cardcoded? Shouldn't this function follow proper pagination implementation and not hardcode values?

Reply:
updated